### PR TITLE
Update error codes cheat sheet for the new "Safer Intent" rules.

### PR DESCRIPTION
**Pull request by jdcormie**
https://developer.android.com/about/versions/15/behavior-changes-15#safer-intents

### CODE REVIEW

**Comment by jdcormie** [comment]
Sorry for the messy diff but IIRC but this table needs rowspans and so the only way to express it was raw HTML. The new case has to go at the beginning to fit in with the other cases that map to UNIMPLEMENTED

**Review by ejona86 — APPROVED** [review]
Yeah, having that many cases does make things annoying on insert. But choices are limited.

File: documentation/android-binderchannel-status-codes.md [hunk 1]
```diff
@@ -326,14 +333,15 @@ According to a review of the AOSP source code, there are in fact several cases:
...
 
 Status code mapping: **UNIMPLEMENTED**
 
 (1) and (2) are interesting new possibilities unique to on-device RPC. (1) is straightforward and the most likely cause of (2) is that the user has an old version of the server app installed that predates its gRPC integration. Many clients will want to handle these cases, likely by directing the user to the app store in order to install/upgrade the server.
 
 Unfortunately `UNIMPLEMENTED` doesn’t capture (3) but none of the other canonical status codes do either and we expect this case to be extremely rare.
 
-(4) is intentially indistinguishable from (1) by Android design so we can't handle it differently. However, as a client manifest error, it's not something reasonable apps would handle at runtime anyway.
+(4) and (5) are intentially indistinguishable from (1) by Android design so we can't handle them differently. However, as an error in its own manifest, (4) isn't something a reasonable client would handle at runtime anyway. (5) is an error in the server manifest and so just like the other cases, best error handling it is to send the user to the app store to look for an updated version than hopefully includes a fix.
```
Comment (groakley) [inline documentation/android-binderchannel-status-codes.md:344]: Wording nits: 

* `%s/best error handling it is to/best error handling is to/` or maybe `the best practice for error handling is...`
* `%s/version than hopefully/version that hopefully/`

Comment (jdcormie) [reply documentation/android-binderchannel-status-codes.md]: fixed thank you!

**Pull request merged**